### PR TITLE
[BUG][JAVA] Prevent generating "pattern" and "size" to ENUM

### PR DESCRIPTION
**Pull request by jorgerod**
Closes: #18477

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09) @martin-mfg (2023/08)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java [hunk 1]
```diff
@@ -741,6 +741,17 @@ public static boolean isURISchema(Schema schema) {
                 && URI_FORMAT.equals(schema.getFormat());
     }
 
+    private static boolean isEnumSchema(final Schema<?> schema) {
+        // MyEnum:
+        //   type: string
+        //   enum:
+        //   - ENUM_1
+        //   - ENUM_2
+        return SchemaTypeUtil.STRING_TYPE.equals(schema.getType()) // type: string
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java:750]: what about enum of integers?

Comment (jorgerod) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java]: You are right. It has to be for all types of enum. I change it

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java [hunk 2]
```diff
@@ -741,6 +741,16 @@ public static boolean isURISchema(Schema schema) {
                 && URI_FORMAT.equals(schema.getFormat());
     }
 
+    private static boolean isEnumSchema(final Schema<?> schema) {
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java:744]: can you please refactor https://github.com/OpenAPITools/openapi-generator/blob/e8eee2fd84081935bedd1fcabf73fc53fb79d111/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java#L1384 to use this new function?

Comment (jorgerod) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java:725]: Done

**Comment by wing328** [comment]
@jorgerod do you mind filing a new PR with just 1 commit as we prefer cleaner commit history? (19 out of 21 commits in this PR do not seem to be relevant to the change)

**Comment by jorgerod** [comment]
> @jorgerod do you mind filing a new PR with just 1 commit as we prefer cleaner commit history? (19 out of 21 commits in this PR do not seem to be relevant to the change)

Yes, I will, but haven't you thought about mixing with "squash and merge" instead of "merge"?

**Comment by wing328** [comment]
we've been doing that (squash and merge) for years.

cleaner commits help everyone and it's something we ask contributors to do recently (and everyone is ok with that so far)

**Comment by jorgerod** [comment]
Deleted in favor of: 
- https://github.com/OpenAPITools/openapi-generator/pull/18658

**Pull request closed without merging**